tauri-app/src-tauri/src/portapi/types.rs

```diff
@@ -0,0 +1,199 @@
+
+use super::message::*;
+
+/// Request is a strongly typed request message
+/// that can be converted to a `portapi::message::Message`
+/// object for further use by the client (`portapi::client::PortAPI`).
+#[derive(PartialEq, Debug)]
+pub enum Request {
+    Get(String),
+    Query(String),
+    Subscribe(String),
+    QuerySubscribe(String),
+    Create(String, Payload),
+    Update(String, Payload),
+    Insert(String, Payload),
+    Delete(String),
+    Cancel,
+}
+
+/// Implementation to convert a internal `portapi::message::Message` to a valid
+/// `Request` variant.
+/// 
+/// Any error returned will be of type `portapi::message::MessageError`.
+impl std::convert::TryFrom<Message> for Request {
+    type Error = MessageError;
+
+    fn try_from(value: Message) -> Result<Self, Self::Error> {
+        match value.cmd.as_str() {
+            "get" => {
+                let key = value.key.ok_or(MessageError::MissingKey)?;
+                Ok(Request::Get(key))
+            },
+            "query" => {
+                let key = value.key.ok_or(MessageError::MissingKey)?;
+                Ok(Request::Query(key))
+            },
+            "sub" => {
+                let key = value.key.ok_or(MessageError::MissingKey)?;
+                Ok(Request::Subscribe(key))
+            },
+            "qsub" => {
+                let key = value.key.ok_or(MessageError::MissingKey)?;
+                Ok(Request::QuerySubscribe(key))
+            },
+            "create" => {
+                let key = value.key.ok_or(MessageError::MissingKey)?;
+                let payload = value.payload.ok_or(MessageError::MissingPayload)?;
+                Ok(Request::Create(key, payload))
+            },
```

Comment:
Retrieving those values could be methods on the `Message` struct. This might simplify/short this code a lot